### PR TITLE
Initial code commit

### PR DESCRIPTION
**Pull request by elifoster**
This adds all Gemfiles, gemspecs, and code mentioned in #1 that are necessary for the new charts in ghuls-apps/ghuls-lib#5.

When merged, I will tag the commit as the new version, build, and push to rubygems. Then we can simply update the CLI and web to use this new gem.

In this commit I used a combination of open-uri and nokogiri (yikes, another dependency) in order to quickly parse the GitHub user page. Pretty much every decision I made should be pretty easily understood in the documentation.

~~The only current issue is that it does not work when the user does not actually exist. This is a problem that I will fix in a future release.~~ Edit: See @c84c24d

@Strikingwolf please perform a review of this so that we can merge with master and push it out.

### CODE REVIEW

File: lib/github/calendar.rb [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    extend self
+    using StringUtility
+
+    # Gets the total number of contributions for the past year.
+    # @param user [String] The username to get this data for.
+    # @return [Int] An integer value of their total contributions this year.
+    # @return [Nil] If the user cannot be found.
+    def get_total_year(user)
+      source = get_page_source(user)
+      return nil if source.nil?
```
Comment (FayeAlephNil) [inline lib/github/calendar.rb]: Would be better to either throw a custom NoUserFound error or to wrap this up in something

File: lib/github/calendar.rb [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      string.to_i_separated
+    end
+
+    # Gets the longest contribution streak (in days) that the user has had.
+    # @param user [String] See #get_total_year
+    # @return [Int] The user's longest contribution streak in days.
+    # @return [Nil] See #get_total_year
+    def get_longest_streak(user)
+      source = get_page_source(user)
+      return nil if source.nil?
```
Comment (FayeAlephNil) [inline lib/github/calendar.rb]: See above

File: lib/github/calendar.rb [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      string.to_i_separated
+    end
+
+    # Gets the current contribution streak (in days) that the user is in.
+    # @param user [String] See #get_total_year
+    # @return [Int] The user's current contribution streak in days.
+    # @return [Nil] See #get_total_year
+    def get_current_streak(user)
+      source = get_page_source(user)
+      return nil if source.nil?
```
Comment (FayeAlephNil) [inline lib/github/calendar.rb]: See above

**Comment by FayeAlephNil** [comment]
Wherever you're returning nil is not nice :laughing:

**Comment by elifoster** [comment]
Hm. I was going to write a thing about how returning nil isn't actually bad in this case, but [an article](http://www.yegor256.com/2014/05/13/why-null-is-bad.html) changed my mind. 

Originally I was thinking in a "we expect the user to not exist because users may not exist on GitHub". However...

It is actually an error if the user doesn't exist, because the application expects the user to exist. However, for our needs (CLI and Web), we will have to rescue from this exception. This isn't a huge deal, and isn't much more coding than checking for nil.

**Comment by elifoster** [comment]
@Strikingwolf is that sufficient?

**Comment by FayeAlephNil** [comment]
Looks good. As far as the other apps are concerned I would create a general method to get data with some method, a lambda to execute on the data, and that displays a "Data not found" thing when the exception is thrown. Then just use that method in all the application methods and it will do all the exception handling automatically

**Pull request merged**